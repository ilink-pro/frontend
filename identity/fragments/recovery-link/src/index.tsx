export * from './recovery-link.component'
