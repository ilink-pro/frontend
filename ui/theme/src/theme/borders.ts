const borders = {
  soft: '1px solid #E3DEF0',
}

export { borders }
