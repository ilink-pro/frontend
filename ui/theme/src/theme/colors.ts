const colors = {
  background: {
    white: 'rgb(255,255,255)',
    lightPurple: 'rgb(251,250,255)',
    lightPurpleAccent: 'rgb(219,216,238)',
    purpleAccent: 'rgb(116,97,240)',
  },
  text: {
    primary: 'rgb(54,52,61)',
    accent: 'rgb(116,97,240)',
    secondary: 'rgb(153,147,189)',
    red: 'rgb(220,103,103)',
    gray: 'rgb(155,164,167)',
    white: 'rgb(255,255,255)',
  },
  button: {
    secondary: {
      default: {
        background: 'rgb(255,255,255)',
        border: 'rgb(219,216,238)',
        font: 'rgb(116,97,240)',
      },
      hover: {
        background: 'rgb(255,255,255)',
        border: 'rgb(116,97,240)',
        font: 'rgb(116,97,240)',
      },
      pressed: {
        background: 'rgb(255,255,255)',
        border: 'rgb(231,228,251)',
        font: 'rgb(116,97,240)',
      },
      disabled: {
        background: 'rgb(255,255,255)',
        border: 'rgb(219,216,238)',
        font: 'rgb(153,147,189)',
      },
    },
  },
  select: {
    menu: {
      border: 'rgb(219,216,238)',
    },
    item: {
      default: {
        background: 'rgb(255,255,255)',
      },
      hover: {
        background: 'rgb(247,246,254)',
      },
    },
  },
}

export { colors }
