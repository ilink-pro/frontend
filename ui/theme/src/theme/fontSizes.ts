export const fontSizes = {
  small: 12,
  semiRegular: 14,
  default: 16,
  increased: 18,
  semiGiant: 24,
  semiBig: 28,
  huge: 32,
  big: 36,
  giant: 40,
  colossal: 64,
}
