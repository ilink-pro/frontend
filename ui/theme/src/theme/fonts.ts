const fonts = {
  primary: 'Khula',
}

export { fonts }
