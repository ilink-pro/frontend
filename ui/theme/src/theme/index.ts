export * from './fontWeights'
export * from './lineHeights'
export * from './fontSizes'
export * from './colors'
export * from './fonts'
export * from './radii'
export * from './shadows'
export * from './borders'
