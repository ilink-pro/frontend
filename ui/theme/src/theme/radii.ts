const radii = {
  normal: 8,
  semiMedium: 12,
  medium: 16,
  full: '100%',
}

export { radii }
