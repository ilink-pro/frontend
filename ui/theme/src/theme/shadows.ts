const shadows = {
  bottom: '0px 4px 24px rgba(0, 0, 0, 0.05)',
  light: '0px 7px 16px rgba(136, 165, 191, 0.12)',
}

export { shadows }
