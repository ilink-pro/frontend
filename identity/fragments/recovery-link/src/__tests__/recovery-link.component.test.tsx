/**
 * @jest-environment jsdom
 */

import { render }        from '@testing-library/react'

import React             from 'react'
import { IntlProvider }  from 'react-intl'

import { ThemeProvider } from '@ui/theme'

import { RecoveryLink }  from '../recovery-link.component'

describe('identity', () => {
  describe('recovery-link', () => {
    describe('recovery-link.component', () => {
      it('should match the latest render snapshot', () => {
        const { asFragment } = render(
          <IntlProvider locale='ru' messages={{}}>
            <ThemeProvider>
              <RecoveryLink />
            </ThemeProvider>
          </IntlProvider>
        )

        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})
