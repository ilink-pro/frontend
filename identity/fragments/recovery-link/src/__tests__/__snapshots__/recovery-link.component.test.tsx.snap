// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`identity recovery-link recovery-link.component should match the latest render snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

.emotion-1 {
  box-sizing: border-box;
  height: 80px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 12px;
  -ms-flex-preferred-size: 12px;
  flex-basis: 12px;
}

.emotion-3 {
  box-sizing: border-box;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (min-width: 40em) {
  .emotion-3 {
    width: 100%;
  }
}

@media screen and (min-width: 52em) {
  .emotion-3 {
    width: 240px;
  }
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  word-break: break-word;
  cursor: pointer;
  outline: none;
  padding: 0;
  border: 1px solid transparent;
  height: 48px;
  font-size: 14px;
  font-weight: 600;
  font-family: Khula;
  padding-left: 24px;
  padding-right: 24px;
  border-radius: 6px;
  background-color: black;
  border-color: black;
  color: white;
}

<div
    class="emotion-0"
    display="flex"
    height="100%"
  >
    <div
      class="emotion-1"
      display="flex"
      height="80"
      width="100%"
    >
      <div
        class="emotion-2"
        display="flex"
      />
    </div>
    <div
      class="emotion-3"
      display="flex"
      width="100%,100%,240"
    >
      <button
        class="emotion-4"
      >
        Send code
      </button>
    </div>
  </div>
</DocumentFragment>
`;
